### DISCUSSION TITLE
How to add the source of custom trained ner component?

### DISCUSSION THREAD

**Discussion by shrinidhin** [explosion/spaCy#10160] [Help: Coding & Implementations] [answered]
Hi, I am trying to create an entity + sentiment analysis project. I have a custom trained NER component. I need to add the source of this custom trained NER component into my other spacy pipeline. I tried going through the documentation but don't understand what exactly should I include in the config file in the `source = ${paths.ner_model}` section:

```
[components.ner]
factory = "ner"
source = ${paths.ner_model}
incorrect_spans_key = null
moves = null
scorer = {"@scorers":"spacy.ner_scorer.v1"}
update_with_oracle_cut_size = 100
```
I have trained and obtained the model_best and model_last, saved in the specified folder. Do I have to add a custom factory name to my custom NER model in the config file and add that to my second pipeline?

**Comment by thomashacker** [comment, marked as answer]
Hello, 
If you haven't renamed the NER component in your trained model, you can source the component this way:

```
[components.ner]
source = ${paths.ner_model}
```
If you renamed the NER component then you'll need to add the `component` variable (not `factory`) 
```
[components.ner]
component = "<component name>"
source = ${paths.ner_model}
```

For the `${paths.ner_model}` variable you have to set the path to either `model-best` or `model-last` directory

**Reply by shrinidhin** [reply]
Great!Thank you very much!
